src/Modules/OrchardCore.Commerce/Extensions/ProductInventoryProviderExtensions.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Abstractions;
+
+public static class ProductInventoryProviderExtensions
+{
+    public static async Task<bool> IsAvailableAsync(this IProductInventoryProvider provider, string sku) =>
```

Comment:
You should never work with a single provider. Always use `IEnumerable<IProductInventoryProvider>`. You can get the currently applicable provider using `var provider = await providers.GetFirstApplicableProviderAsync(model)`.